tests/integration/helm/helm_upgrade_test.go

```diff
@@ -0,0 +1,170 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework/image"
+
+	"istio.io/istio/pkg/test/env"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/helm"
+)
+
+var (
+	// previousChartPath is path of Helm charts for previous Istio deployments.
+	previousChartPath        = filepath.Join(env.IstioSrc, "tests/integration/helm/testdata/")
+	previousSupportedVersion = "1.8.1"
+
+	istioDockerHub = "docker.io/istio"
+)
+
+// TestDefaultInPlaceUpgrades tests Istio installation using Helm with default options
+func TestDefaultInPlaceUpgrades(t *testing.T) {
+	framework.
+		NewTest(t).
+		Features("installation.helm.default.upgrade").
+		Run(func(ctx framework.TestContext) {
+			workDir, err := ctx.CreateTmpDirectory("helm-upgrade-test")
+			if err != nil {
+				t.Fatal("failed to create test directory")
+			}
+
+			cs := ctx.Clusters().Default().(*kube.Cluster)
+			h := helm.New(cs.Filename(), filepath.Join(previousChartPath, previousSupportedVersion))
+			overrideValuesStr := `
+global:
+  hub: %s
+  tag: %s
+`
+			overrideValues := fmt.Sprintf(overrideValuesStr, istioDockerHub, previousSupportedVersion)
+			overrideValuesFile := filepath.Join(workDir, "values.yaml")
+			if err := ioutil.WriteFile(overrideValuesFile, []byte(overrideValues), os.ModePerm); err != nil {
+				t.Fatalf("failed to write iop cr file: %v", err)
+			}
+			ctx.WhenDone(func() error {
+				// only need to do call this once as helm doesn't need to remove
+				// all versions
+				return deleteIstio(ctx, cs, h)
+			})
+			installIstio(ctx, cs, h, overrideValuesFile)
+
+			verifyInstallation(t, ctx, cs)
+
+			s, err := image.SettingsFromCommandLine()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			overrideValues = fmt.Sprintf(overrideValuesStr, s.Hub, s.Tag)
+			overrideValuesFile = filepath.Join(workDir, "values-upgrade.yaml")
+			if err := ioutil.WriteFile(overrideValuesFile, []byte(overrideValues), os.ModePerm); err != nil {
+				t.Fatalf("failed to write iop cr file: %v", err)
+			}
+			upgradeIstio(ctx, h, overrideValuesFile)
+			verifyInstallation(t, ctx, cs)
+		})
+
+}
+
+// TestInstallWithFirstPartyJwt tests Istio installation using Helm
+// with first-party-jwt enabled
+func TestUpgradesWithFirstPartyJwt(t *testing.T) {
```

Comment:
nit: looks like a lot of identical logic between the tests (same except for the `jwtPolicy: first-party-jwt`, right?)

Reply:
Cleaned it up a bit. Hopefully it cleans it up. Good suggestion.